### PR TITLE
📦 Build binaries on tag pushes

### LINKED ISSUE #33: Create pipelines for creating drafts and releasing
We need to create pipelines to properly compile the binaries before we can put them in the store

### PR DESCRIPTION
**Pull request by heytherewill**
Closes #33 

This creates the binaries and adds them to the action. Further work for uploading them to the package manager registries is to be done in #31.

Do note that I'm not creating releases. IMO these should be created manually so we can add the changelog (or maybe we can have a rolling changelog that we add things on each commit? We need to think this through)

### CODE REVIEW

**Comment by shantanuraj** [comment]
I think we can explore the idea of creating [draft releases](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) on each merge to master,
so we'd have always have a draft at hand and would only need to publish it when we're ready.

But yeah that can be a separate PR.

File: .github/workflows/tag.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release
+
+      - name: Upload Build Artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: Binary
+          path: target/release/toggl.exe
```
Comment (shantanuraj) [inline .github/workflows/tag.yml:42]: Would this only store the artifcat from the windows output, cause the mac and linux binaries would lack the toggl extension.

We can probably use wildcards here
```yml
path: target/release/toggl*
```

Comment (heytherewill) [reply .github/workflows/tag.yml:42]: Good catch

**Review by shantanuraj — APPROVED** [review]
LGTM 🪨 

From test repo
image
Screenshot 2022-04-18 at 16 30 55

**Pull request merged**